src/pepr/operator/controllers/keycloak/authservice/authorization-policy.ts

```diff
@@ -0,0 +1,164 @@
+import { K8s, Log } from "pepr";
+import { UDSConfig } from "../../../../config";
+import { Action, AuthorizationPolicy, RequestAuthentication, UDSPackage } from "../../../crd";
+import { getOwnerRef } from "../../utils";
+import { Action as AuthServiceAction, AuthServiceEvent } from "./types";
+
+function authserviceAuthorizationPolicy(
+  labelSelector: { [key: string]: string },
+  name: string,
+  namespace: string,
+): AuthorizationPolicy {
+  return {
+    kind: "AuthorizationPolicy",
+    metadata: {
+      name: `${name}-authservice`,
+      namespace,
+    },
+    spec: {
+      action: Action.Custom,
+      provider: {
+        name: "authservice",
+      },
+      rules: [
+        {
+          when: [
+            {
+              key: "request.headers[authorization]",
+              notValues: ["*"],
+            },
+          ],
+        },
+      ],
+      selector: {
+        matchLabels: labelSelector,
+      },
+    },
+  };
+}
+
+function jwtAuthZAuthorizationPolicy(
+  labelSelector: { [key: string]: string },
+  name: string,
+  namespace: string,
+): AuthorizationPolicy {
+  return {
+    kind: "AuthorizationPolicy",
+    metadata: {
+      name: `${name}-jwt-authz`,
+      namespace,
+    },
+    spec: {
+      selector: {
+        matchLabels: labelSelector,
+      },
+      rules: [
+        {
+          from: [
+            {
+              source: {
+                requestPrincipals: [`https://sso.${UDSConfig.domain}/realms/uds/*`],
+              },
+            },
+          ],
+        },
+      ],
+    },
+  };
+}
+
+function authNRequestAuthentication(
+  labelSelector: { [key: string]: string },
+  name: string,
+  namespace: string,
+): RequestAuthentication {
+  return {
+    kind: "RequestAuthentication",
+    metadata: {
+      name: `${name}-jwt-authn`,
+      namespace,
+    },
+    spec: {
+      jwtRules: [
+        {
+          forwardOriginalToken: true,
+          issuer: `https://sso.${UDSConfig.domain}/realms/uds`,
+          jwksUri: `https://sso.${UDSConfig.domain}/realms/uds/protocol/openid-connect/certs`,
+        },
+      ],
+      selector: {
+        matchLabels: labelSelector,
+      },
+    },
+  };
+}
+
+async function updatePolicy(
+  event: AuthServiceEvent,
+  labelSelector: { [key: string]: string },
+  pkg: UDSPackage,
+) {
+  const operation = event.action === AuthServiceAction.Add ? "Apply" : "Delete";
```

Comment:
This is a typesafe and future proof equivalent (it will throw type errors if there are new items added to the Enum):
```suggestion
  const operation = {[AuthServiceAction.Add]: "Apply", [AuthServiceAction.Remove]: "Delete"}[event.action];
```